community/modules/scheduler/schedmd-slurm-gcp-v6-controller/modules/slurm_files/scripts/util.py

```diff
@@ -1754,13 +1756,13 @@ def nodeset_reservation(self, nodeset: object) -> Optional[ReservationDetails]:
         assert len(zones) == 1, "Only single zone is supported if using a reservation"
         zone = zones[0]
 
-        try:
-            _, project, _, name = nodeset.reservation_name.split("/")
-        except ValueError:
+        regex = re.compile(r'^projects/(?P<project>[^/]+)/reservations/(?P<reservation>[^/]+)(/reservationBlocks/(?P<block>[^/]+))?$')
+        if not (match := regex.match(nodeset.reservation_name)):
             raise ValueError(
-                f"Invalid reservation name: '{nodeset.reservation_name}', expected format is 'projects/PROJECT/reservations/NAME'"
+                f"Invalid reservation name: '{nodeset.reservation_name}', expected format is 'projects/PROJECT/reservations/NAME[/reservationBlocks/BLOCK]'"
```

Comment:
Is the [ character next to NAME supposed to be / instead?

Like .../reservations/NAME/[reservationBlocks/BLOCK] instead of .../reservations/NAME[/reservationBlocks/BLOCK]

Reply:
No, / is part of optional suffix